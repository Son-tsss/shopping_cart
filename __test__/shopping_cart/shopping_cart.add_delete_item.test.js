import ShoppingCart from '../../src/shopping_cart'

it('should add item to the items list', ()=>{
    const shoppingCart = new ShoppingCart

    const newItem  = {
        id: "i1",
        price: 100
    }

    shoppingCart.addItem(newItem)

    const {currentState: real} = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})

it('should increment existing item in the items list instead of adding', ()=>{
    const shoppingCart = new ShoppingCart

    const newItem  = {
        id: "i1",
        price: 100
    }

    shoppingCart.addItem(newItem)
    shoppingCart.addItem(newItem)

    const {currentState: real} = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})

it('should delete existing item from the items ', ()=>{
    const shoppingCart = new ShoppingCart

    const newItem  = {
        id: "i1",
        price: 100
    }

    shoppingCart.addItem(newItem)
    shoppingCart.deleteItem(newItem.id)

    const {currentState: real} = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})

it('deleting not existing item should change nothing ', ()=>{
    const shoppingCart = new ShoppingCart

    const newItem  = {
        id: "i1",
        price: 100
    }

    shoppingCart.addItem(newItem)
    shoppingCart.deleteItem("i3")

    const {currentState: real} = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})

