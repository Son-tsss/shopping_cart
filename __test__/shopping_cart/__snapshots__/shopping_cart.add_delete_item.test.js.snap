// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleting not existing item should change nothing  1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 1,
      "discount": 1,
      "id": "i1",
      "price": 100,
    },
  ],
}
`;

exports[`should add item to the items list 1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 1,
      "discount": 1,
      "id": "i1",
      "price": 100,
    },
  ],
}
`;

exports[`should delete existing item from the items  1`] = `
Object {
  "discount": 1,
  "items": Array [],
}
`;

exports[`should increment existing item in the items list instead of adding 1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 2,
      "discount": 1,
      "id": "i1",
      "price": 100,
    },
  ],
}
`;
